src/lib.rs

```diff
@@ -139,15 +139,16 @@ impl<T: Clone> Grid<T> {
         if rows < 1 || cols < 1 {
             panic!("Grid size of rows and columns must be greater than zero.");
         }
-        Grid {
```

Comment:
Is there a reason for changing this? I don't see a benefit here.

Reply:
It allows using `new` with non-`Clone` types. For example, `Option<T>` in a generic context would require `T: Clone` otherwise, which is mildly ugly if all you need is `None`.

Reply:
Hm... Still don't get it. Sure the whole PR makes sense with not requiring `T: Clone` for most operations, though what did the change from: 

```
Grid { 
   data: vec![T::default(); rows * cols],
   cols,
   rows,
}
```

to

```
let mut data = Vec::new();
data.resize_with(rows * cols, T::default);
Grid { data, cols, rows }
```

do? There is no difference I would assume?

Reply:
`vec!` [requires Clone](https://doc.rust-lang.org/std/macro.vec.html) (unless the size is zero, it seems), so getting rid of it is required to drop the bound.

After reading the small print for `vec!`: there _is_ a difference. For an [`Rc<T> where T: Default`](https://doc.rust-lang.org/std/rc/struct.Rc.html#impl-Default) the current version would clone `Rc`s and they all would point to the same `T`. The one from the PR would create many independent `Rc`s.

Reply:
Thanks for the explanation. Your change makes total sense.